### ISSUE TITLE
Memory leak introduced by "Partially split server and client"

### ISSUE THREAD

**Issue by packer-** [TheRegulars/xonplaces#1]
Commit 37ccf888 introduced a memory leak in `xonplaces/src/console.c:600`.

---

## Steps to reproduce:

*With issue:*

- Checkout commit 37ccf888, compile and run:
`./all run dedicated +developer 1 +defer 5 memstats`
- Expected result:
```
283 memory pools, totalling 319321791 bytes (304.529MB)
total allocated size: 329181015 bytes (313.931MB)
Memory pool 0x55ba75602ee0 has sprung a leak totalling 32768 bytes (0.031MB)!  Listing contents...
     65536 bytes allocated at …/xonplaces/src/console.c:600
```

*Without issue:*

- Checkout previous commit `git checkout HEAD~1` (9b70e6ae)
```
Previous HEAD position was 37ccf888 Partially split server and client
HEAD is now at 9b70e6ae Do not link server version with png librar
```
- Compile and run:
`./all run dedicated +developer 1 +defer 5 memstats`
- Expected result (no leak in `console.c`):
```
283 memory pools, totalling 303800154 bytes (289.726MB)
total allocated size: 313281242 bytes (298.768MB)
```
---

The memory leak grows with time:
```
Memory pool 0x55c2ad9d1eb0 has sprung a leak totalling 67108864 bytes (64.000MB)!  Listing contents...
67108864 bytes allocated at …/xonplaces/src/console.c:600
```

Note: I copied the compiled xonplaces binary into my Xonotic checkout.

**Comment by bacher09** [comment]
Yeah, you are completely right, thanks for you work, it's outstanding bug report.

Basically after 37ccf88 server got empty [SCR_BeginLoadingPlaque](https://github.com/TheRegulars/xonplaces/blob/master/src/cl_screen.c#L687) function, and that function called [Log_Start](https://github.com/TheRegulars/xonplaces/blob/master/src/console.c#L545) function which cleaned `logqueue` (leaking buffer). 

I'll try to fix this today.

**Comment by bacher09** [comment]
This should be fixed by 467c278. Reopen issue if you still can reproduce it.
